### PR TITLE
Sort armor UI item information pane cleanup

### LINKED ISSUE #56055: Bugs with "Wearing multiple items..." info in Sort Armor menu
### Describe the bug

"Wearing multiple items..." info in Sort Armor menu is showing the same body parts several times. Also this info hides some body part when enumerating body parts.

### Steps to reproduce

Wear two sweatshirts and open Sort Armor menu, check middle pane with "Wearing multiple items..." info.

### Expected behavior

No duplicating and no invisible body parts in aforementioned info text.

### Screenshots

изображение

### Versions and configuration

Windows 10 x64, 0.F-6755.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ZeroInternalReflection**
#### Summary
Interface "Sort Armor UI item information pane cleanup"

#### Purpose of change
The middle pane of the Sort Armor UI (`+`) has several display glitches and navigational issues, particularly on small screens and in non-English languages.  These include:
1] The warning about wearing multiple items on the same layer will list the same body part multiple times (#56055)
1  Sort Armor -- 116x54 - Glitchy multiple item listing part 2
fixes #56055
2] On small screens, [Protection] values for the selected item are not displayed.  Since the middle pane does not scroll, it is then impossible to see that information on this UI on small screens.  This also prevents the player from seeing the warning about improperly layered clothing (i.e. the purpose of this screen)
2 -1 Sort Armor -- 80x33 - Run out of space for protection values
2 -2 Sort Armor -- 80x32 - Protection values inaccessible
3] On certain languages, the [Properties]/[Protection] values will wrap around and print at the beginning of the next line.
3  Sort Armor -- 80x47 -- Russian - Item properties wrap around
4] [Protection] values that give Best/Median/Worst values, such as "pair of combat boots" when looking at the feet, will wrap around and overprint in most languages
4  Sort Armor -- 80x35 -- Item properties wrap around
5] While there's a check to see if there's room for [Protection] values, there is no equivalent check for item flags or layering warnings, causing those to print over each other at the bottom of the section
5 -1 Sort Armor -- 80x39 - Glitchy multiple item listing
6] The help screen (Which is very neat, and should be more common) will not catch if the user has custom keybindings for scrolling
6  Sort Armor -- 80x24 - Help screen - Missing left+right, misses rebind of up+down
7] The Encumbrance and Warmth menu does not have space between the body part name and encumbrance when encumbrance is >= 100

#### Describe the solution
(A bunch of before/after screenshots are included in the "Testing" section below)
1] Update the listing of body parts to filter out duplicates and combine where possible.  The original issue was due to every sublocation being listed, with each sublocation having the same accusative name.
2], 5] Add scrolling to the middle pane (currently `<`/`>`), allowing everything to be printed and accessed.  I've also moved the layering warnings to the top of the pane, since I think that's the most important information for this screen.
3] Allow protection entries to wrap around if necessary
4] Split best/median/worst values into separate lines.  This does, unfortunately, break translation.
6] Update the help menu to use currently-defined keybindings for navigation.  I've also added the keybindings for scrolling the middle pane to the list.  Since this does break translation, I've split the help menu into multiple translation calls so that next time it should be possible to modify a single line without breaking translation on the entire screen.  I've also converted it to "scrollable_text", so that additional help text (such as information on Warmth - #49806) can be added and still be accessible on small screens.
Sort Armor - Help screen -- 80x24
Sort Armor - Help screen  -- Large
7] Update the Encumbrance and Warmth table to try to centre the encumbrance values.  Note: this also has an impact on the player status screen (`@`):
Character status - Encumbrance and Warmth

#### Describe alternatives you've considered
- Reworking this screen completely.  For instance, do we _really_ need the item listing in both the left and right panes?
Decided to hold off for now.  If certain aspects get more complicated, then the duplicate listing will almost certainly need to be cut for screen real estate
- Cutting down on information presented in this screen.  Is there a better place to compare armour protection values?
At this point, this screen seems to be the best option for trying to sort out a balance of protection, warmth, and encumbrance.  Storage could probably be split out to the new pocket management screen.

#### Testing
Tested the updated screen at a variety of different screen sizes in several different languages:

Scrolling example:

https://user-images.githubusercontent.com/89038572/174090063-bb799aaf-3b69-43ef-b110-79e4a743fe13.mp4

<details><summary>English, 80x24</summary>

Before:
Sort Armor -- English -- 80x24
After:
Sort Armor -- English -- 80x24

</details>
<details><summary>English, 115x35</summary>

Before:
Sort Armor -- 115x35
After:
Sort Armor -- 115x35
</details>

<details><summary>English, 240x63</summary>

Before:
Sort Armor -- English -- 240x63
After:
Sort Armor -- English -- 240x63
</details>

<details><summary>Chinese (Taiwan), 80x24</summary>

Before:
Sort Armor -- Traditional Chinese -- 80x24
After:
Sort Armor -- Traditional Chinese -- 80x24
</details>

<details><summary>Chinese (Taiwan), 240x63</summary>

Before:
Sort Armor -- Traditional Chinese -- Large
After:
Sort Armor -- Traditional Chinese -- Large
</details>

<details><summary>Spanish (Argentina), 80x35</summary>

Before:
Sort Armor -- Spanish (Argentina) -- 80x35
After:
Sort Armor -- Spanish (Argentina) -- 80x35
</details>

<details><summary>Polish, 85x40</summary>

Before:
Sort Armor -- Polish -- 85x40
After:
Sort Armor -- Polish -- 85x40
</details>

#### Additional context
- The header is next on the to-do list
- There are several more tweaks I'm tempted to make (such as dynamically shrinking the Encumbrance and Warmth panel based on the body parts listed), but this PR is already dramatically more complicated than the previous two tweaking this UI, and they're "this would look nicer" rather than "this allows the player to see all relevant information"
- Most of these changes should be independent, and if splitting this PR would help review, it _should_ be relatively straightforward to split some of these changes out
- There's a helper function (name_and_value) used in this pane (and only this pane)  that I wish I'd realized was there when working on the other panes
- There's a minor display glitch at the top right of the Encumbrance and Warmth pane where it meets the right pane scrollbar.  I don't see any way to fix it without adding a complicated exception to the scrollbar code.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Is it possible to make it print `Worst / Median / Best` values in one line if there's enough place for it, and print each value on its own line if there's not enough free space?

File: src/armor_layers.cpp [hunk 1]
```diff
@@ -274,103 +290,94 @@ std::vector<std::string> clothing_protection( const item &worn_item, const int w
...
+        { flag_SUN_GLASSES, "It keeps the sun out of your eyes." },
+        { flag_WATERPROOF, "It is waterproof." },
+        { flag_WATER_FRIENDLY, "It is water friendly." },
+        { flag_FANCY, "It looks fancy." },
+        { flag_SUPER_FANCY, "It looks really fancy." },
+        { flag_FLOTATION, "You will not drown today." },
+        { flag_OVERSIZE, "It is very bulky." },
+        { flag_SWIM_GOGGLES, "It helps you to see clearly underwater." },
+        { flag_SEMITANGIBLE, "It can occupy the same space as other things." }
+    };
```
[the remark this answers is not in the document]
Comment (BrettDong) [reply src/armor_layers.cpp:373]: Yes, the `xgettext` program does not know these strings need to be translated.

**Comment by ZeroInternalReflection** [comment]
> Is it possible to make it print Worst / Median / Best values in one line if there's enough place for it, and print each value on its own line if there's not enough free space?

It's definitely possible.  I'll need to play around with clothing_protection() to see how complicated it would be.  Maybe I can pull assembly of that out as a separate function, rather than doing special cases for bash/cut/projectile...

**Comment by ZeroInternalReflection** [comment]
> Is it possible to make it print Worst / Median / Best values in one line if there's enough place for it, and print each value on its own line if there's not enough free space?

Is something like this what you were imagining?

https://user-images.githubusercontent.com/89038572/174194114-83b933f0-1b16-4499-8b43-41a6fb56d06a.mp4

I still need to do some more testing and cleanup, but thus far it hasn't been too complicated.  It's a little twitchy at the switchover point since the pane widths have some rounding involved, but 'looks nice while actively being resized' is a low priority.

If anyone knows of any items that currently display "Median" values, let me know, since I haven't specifically tested that behaviour.

**Comment by Night-Pryanik** [comment]
> Is something like this what you were imagining?

Yes, exactly! Thank you very much!

**Comment by ZeroInternalReflection** [comment]
Alright, I believe this update fixes:
1] Flagging of translation for clothing descriptions
2] clang-tidy complaining about const in declarations, and 
3] Collapsing of best/median/worst values to a single line if it will fit

I misspoke in my previous post.  I was actually looking for items that had best/worst values, but did _not_ show median values, like the kevlar firesuit:

https://user-images.githubusercontent.com/89038572/174420722-0932e8e3-58ad-41b7-a0a5-8ec3a0ba5854.mp4

File: src/armor_layers.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZeroInternalReflection) [reply src/armor_layers.cpp]: I went with the second example of translate_marker().  update_pot.sh now successfully pulls these strings out, which I believe means this is resolved.

**Pull request merged**